ghc-toolkit/ghc-libdir/include/rts/storage/TSO.h

```diff
@@ -183,6 +183,19 @@ typedef struct StgTSO_ {
     StgWord32 saved_winerror;
 #endif
 
+    // Asterius fields must be at the end of the structure because the GHC
+    // compiler we get Cmm from isn't compiled with this header file but with
+    // the original one. Hence it assumes some offsets in the structure (see the
+    // generated constants by the deriveConstants program).
+    //
+    // Hopefully the size of TSO structure isn't a constant used by the
```

Comment:
To get confidence about how `StgTSO` sizes/offsets are used by the host ghc, grep for `StgTSO` in the [`platformConstants`](https://github.com/tweag/asterius/blob/master/ghc-toolkit/ghc-libdir/platformConstants) file. We're being lucky in this specific case.

Reply:
Yes I did that. I chose the wrong word, I will s/Hopefully/Luckily/